python/ops.py

```diff
@@ -295,37 +434,48 @@ def splat_non_uniform(x, bins):
   method takes NumPy arrays instead of TF tensors, and returns a Numpy array.
 
   Args:
-    x: A 1D vector of values being splatted, in the shape of [batch_size].
+    x: A 2D matrix of values being splatted, in the shape of [batch_size, 1].
     bins: 1D vector, in the shape of [N], where N is the number of bins.
 
   Returns:
-    A 2D matrix with interpolated weights in the shape of [batch_size, M], where
-    each row is the splat weights.
+    f: A 2D matrix with interpolated weights in the shape of [batch_size, N],
+      where each row is the splat weights.
   """
 
-  x = np.asarray(x)
-  bins = np.asarray(bins)
+  N = tf.size(bins)
+  bins = tf.expand_dims(bins, axis=0)
+  batch_size = x.shape[0]
+  if N == 1:
+    return tf.convert_to_tensor(np.ones((batch_size, 1)),
+                                dtype=tf.float32)
 
   # clamps the x into the boundary of bins
-  if (bins.ndim != 1 or x.ndim != 1 or bins.shape[0] <= 1 or x.shape[0] < 1):
-    raise ValueError('Input and output parameters does not meet requirement: '
-                     'x.shape={}, bin.shape={}'.format(x.shape, bins.shape))
-
-  if bins.shape[0] == 1:
-    return np.ones((x.shape[0], 1))
-
-  clamped_x = np.clip(x, bins[0], bins[-1])
-  delta_x = clamped_x[:, np.newaxis] - bins
-  idx_lo = np.minimum(
-      delta_x.shape[1] - np.argmax(delta_x[:, ::-1] >= 0., axis=1) - 1,
-      bins.shape[0] - 2)
+  x_clamp = tf.minimum(
+    tf.math.reduce_max(bins),
+    tf.maximum(tf.math.reduce_min(bins), x))
+
+  idx_lo = tf.expand_dims(tf.minimum(tf.cast(tf.math.argmin(
+    tf.abs(tf.transpose(bins - x_clamp))), dtype=tf.int32), N - 2), axis=1)
+
   idx_hi = idx_lo + 1
-  w_hi = (clamped_x - bins[idx_lo]) / (bins[idx_hi] - bins[idx_lo])
-  w_lo = 1 - w_hi
 
-  f = np.zeros((x.shape[0], bins.shape[0]))
-  f[range(x.shape[0]), idx_lo] = w_lo
-  f[range(x.shape[0]), idx_hi] = w_hi
+  low_bin_value = tf.gather(bins[0, :], idx_lo)
+  high_bin_value = tf.gather(bins[0, :], idx_hi)
+
+  w_high = tf.squeeze(
+    (x_clamp - low_bin_value) / (high_bin_value - low_bin_value),
+    axis=1)
+  w_low = 1. - w_high
```

Comment:
The rest of this file uses 1.0 - let's standardize
```suggestion
  w_low = 1.0 - w_high
```

Reply:
Fixed